### PR TITLE
fix fontDir param for Mpdf 7.x

### PR DESCRIPTION
**Pull request by lutek**
https://github.com/yii2tech/html2pdf/issues/17

### CODE REVIEW

**Comment by lutek** [comment]
with fix work fine:

```php
        $defaultFontConfig = (new \Mpdf\Config\FontVariables())->getDefaults();
        $fontData = $defaultFontConfig['fontdata'];

        $path = Yii::getAlias('@app/data/pdf/');
        $targetFile = $path . $model->qr_key . '.pdf';

        $options = [
            'fontDir' => '@app/data/fonts/',
            'fontdata' => $fontData + [
                    'Roboto' => [
                        'R' => 'Roboto-Regular.ttf',
                        'I' => 'Roboto-Italic.ttf',
                        'B' => 'Roboto-Bold.ttf',
                    ]
                ],
            'default_font' => 'Roboto',
        ];


        Yii::$app->html2pdf
            ->convert($html, $options)
            ->saveAs($targetFile);
```

File: src/converters/Mpdf.php [hunk 1]
```diff
@@ -43,20 +43,25 @@ protected function convertInternal($html, $outputFileName, $options)
...
         } else {
             $pdf = new \mPDF($charset, $pageSize);
         }
 
         foreach ($options as $name => $value) {
             $setter = 'Set' . $name;
             if (method_exists($pdf, $setter)) {
-                $pdf->$setter($value);
+                if (is_array($value)) {
+                    call_user_func_array([$pdf, $setter], $value);
```
Comment (klimov-paul) [inline src/converters/Mpdf.php:61]: This is unacceptable: such change may break setup for other setters.

**Review by klimov-paul — CHANGES_REQUESTED** [review]

**Comment by klimov-paul** [comment]
Resolved by commit c68115bfc4016cd4bc287037bfd17d9526100dce

File: src/converters/Mpdf.php (hunk 1 above)
Comment (lutek) [reply src/converters/Mpdf.php:61]: are setters that require array as an argument ex:
https://mpdf.github.io/reference/mpdf-functions/setprotection.html

**Pull request closed without merging**